firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -1689,4 +1697,21 @@ public void testMultipleFieldPatchesOnLocalDocs() {
     assertChanged(doc("foo/bar", 0, map("likes", 1, "stars", 2)).setHasLocalMutations());
     assertContains(doc("foo/bar", 0, map("likes", 1, "stars", 2)).setHasLocalMutations());
   }
+
+  @Test
+  public void testUpdateOnRemoteDocLeadsToUpdateOverlay() {
+    Query query = query("foo");
+    allocateQuery(query);
+
+    applyRemoteEvent(updateRemoteEvent(doc("foo/baz", 10, map("a", 1)), asList(2), emptyList()));
+    applyRemoteEvent(updateRemoteEvent(doc("foo/bar", 20, map()), asList(2), emptyList()));
+    writeMutation(patchMutation("foo/baz", map("b", 2)));
+
+    resetPersistenceStats();
+
+    localStore.executeQuery(query, /* usePreviousResults= */ true);
+    assertRemoteDocumentsRead(/* byKey= */ 0, /* byCollection= */ 2);
+    assertOverlaysRead(/* byKey= */ 0, /* byCollection= */ 1);
+    assertOverlayTypes(keyMap("foo/baz", CountingQueryEngine.OverlayType.Patch));
```

Comment:
for example we'd check:
```
assertOverlays(keyMap("foo/baz", patchMutation("foo/baz", map("b",2))));
```

Reply:
This would need to be `assertOverlays(..., patchMutation("foo/baz", map(...), Precondition.None))` to pass.